gen/mailaliases

```diff
@@ -16,17 +16,33 @@ my $data = perunServicesInit::getHashedHierarchicalData;
 #Constants
 our $A_USER_LOGIN;      *A_USER_LOGIN =        \'urn:perun:user_facility:attribute-def:virt:login';
 our $A_USER_MAIL;       *A_USER_MAIL =         \'urn:perun:user:attribute-def:def:preferredMail';
+our $A_MEMBER_STATUS;   *A_MEMBER_STATUS =     \'urn:perun:member:attribute-def:core:status';
+our $A_ALLOW_EXPIRED;   *A_ALLOW_EXPIRED =     \'urn:perun:resource:attribute-def:def:allowExpiredMembers';
+
+our $STATUS_VALID;                *STATUS_VALID =            \'VALID';
+our $STATUS_EXPIRED;              *STATUS_EXPIRED =          \'EXPIRED';
 
 my $fileName = "$DIRECTORY/$::SERVICE_NAME";
 open FILE,">$fileName" or die "Cannot open $fileName: $! \n";
 
 my %mailByLogin;
-foreach my $memberId ($data->getMemberIdsForFacility()) {
 
-	my $login  = $data->getUserFacilityAttributeValue( member => $memberId, attrName => $A_USER_LOGIN);
-	my $mail  = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_MAIL);
+foreach my $resourceId ( $data->getResourceIds() ) {
+	my $allowExpiredMembers = $data->getResourceAttributeValue( resource => $resourceId, attrName => $A_ALLOW_EXPIRED );
+
+	foreach my $memberId ($data->getMemberIdsForResource(resource => $resourceId)) {
+		my $status = $data->getMemberAttributeValue( member => $memberId, attrName => $A_MEMBER_STATUS );
+
+		if($allowExpiredMembers) {
+			if(($status ne $STATUS_VALID) && ($status ne $STATUS_EXPIRED)) { next; }
+		} else {
+			if($status ne $STATUS_VALID) { next; }
+		}
 
-	$mailByLogin{$login} = $mail;
+		my $login  = $data->getUserFacilityAttributeValue( member => $memberId, attrName => $A_USER_LOGIN);
+		my $mail  = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_MAIL);
+		$mailByLogin{$login} = $mail;
```

Comment:
The code seems to be ok, I would just simplify this part so it is more readable.
Something like this:
```
if (($status eq $STATUS_VALID) || ($status eq $STATUS_EXPIRED && $allowExpiredMembers)) {
    my $login  = $data->getUserFacilityAttributeValue( member => $memberId, attrName => $A_USER_LOGIN);		 
    my $mail  = $data->getUserAttributeValue( member => $memberId, attrName => $A_USER_MAIL);		 
    $mailByLogin{$login} = $mail;
}
```

Reply:
It looks good! I rewrote it.